cedar-drt/fuzz/src/lib.rs

```diff
@@ -294,6 +294,97 @@ pub fn run_val_test(
     }
 }
 
+// API for request validation
+// pub fn new<S: RequestSchema>(
+//     principal: (EntityUID, Option<Loc>),
+//     action: (EntityUID, Option<Loc>),
+//     resource: (EntityUID, Option<Loc>),
+//     context: Context,
+//     schema: Option<&S>,
+//     extensions: Extensions<'_>,
+// ) -> Result<Self, S::Error>
+
+// pub fn new_with_unknowns<S: RequestSchema>(
+//     principal: EntityUIDEntry,
+//     action: EntityUIDEntry,
+//     resource: EntityUIDEntry,
+//     context: Option<Context>,
+//     schema: Option<&S>,
+//     extensions: Extensions<'_>,
+// ) -> Result<Self, S::Error> {
+
+pub fn run_req_val_test(
+    custom_impl: &impl CedarTestImplementation,
+    schema: ValidatorSchema,
+    request: ast::Request,
+    extensions: Extensions<'_>
+) {
+    let (rust_res, rust_auth_dur) =
+        time_function(|| ast::Request::new_with_unknowns(
+            request.principal().clone(),
+            request.action().clone(),
+            request.resource().clone(),
+            request.context().cloned(),
+            Some(&schema),
+            extensions
+        ));
+    info!("{}{}", RUST_AUTH_MSG, rust_auth_dur.as_nanos());
+
+    let definitional_res = custom_impl.validate_request(&schema, &request);
+    match definitional_res {
+        TestResult::Failure(_) => {
+            ;
+        }
+        TestResult::Success(definitional_res) => {
+            if rust_res.is_ok() {
+                assert!(definitional_res.validation_passed(), "Definitional Errors: {:?}\n, Rust output: {:?}", definitional_res.errors, rust_res.unwrap());
+            }
+            else {
+                assert!(!definitional_res.validation_passed(), "Errors: {:?}", definitional_res.errors);
+            }
+        }
+    }
+}
+
+// API for entity validation 
+// pub fn from_entities(
+//     entities: impl IntoIterator<Item = Entity>,
+//     schema: Option<&impl Schema>,
+//     tc_computation: TCComputation,
+//     extensions: Extensions<'_>,
+// ) -> Result<Self>
+pub fn run_ent_val_test(
+    custom_impl: &impl CedarTestImplementation,
+    schema: ValidatorSchema,
+    entities: Entities,
+    extensions: Extensions<'_>
+) {
+    let (rust_res, rust_auth_dur) =
+    time_function(|| Entities::from_entities(
+        entities.iter().cloned(),
+        Some(&cedar_policy_validator::CoreSchema::new(&schema)),
+        TCComputation::ComputeNow, // todo 
```

Comment:
`AlreadyComputed` (or whatever the name is) should be fine here since the entities are coming from an `Entities` object